src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3187,7 +3187,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           def matches(stat: Tree, synt: Tree) = (stat, synt) match {
             // synt is default arg for stat
             case (DefDef(_, statName, _, _, _, _), DefDef(mods, syntName, _, _, _, _)) =>
-              mods.hasDefault && syntName.toString.startsWith(statName.toString)
+              mods.hasDefault && syntName.decodedName.startsWith(statName)
```

Comment:
interestingly this is the line that fixes https://github.com/scala/bug/issues/10343